src/ophys_etl/utils/traces.py

```diff
@@ -1,56 +1,84 @@
 from typing import List
-from functools import partial
 import numpy as np
 from scipy.sparse import coo_matrix
 from scipy.ndimage.filters import median_filter
 
-# Partial for simplifying repeat median filter calls
-medfilt = partial(median_filter, mode='constant')
 
+def nanmedian_filter(input_arr: np.ndarray, filter_length: int) -> np.array:
+    """ 1D median filtering with np.nanmedian
+    Parameters
+    ----------
+    input_arr: np.ndarray
+        1d array of signal
+    filter_length: int
+        Length of the median filter to compute a rolling baseline
+
+    Return
+    ------
+    filtered_trace
+    """
 
-def robust_std(x: np.ndarray) -> float:
+    half_length = int(filter_length/2)
+    # Create 'reflect' traces at the extrema
+    temp_trace = np.concatenate(
+        (
+            np.flip(input_arr[:half_length]),
+            input_arr,
+            np.flip(input_arr[-half_length:]))
+        )
+    filtered_trace = np.zeros_like(input_arr)
+    for i in range(len(input_arr)):
+        median = np.nanmedian(temp_trace[i:i+filter_length])
+        if np.isnan(median) and i > 0:
```

Comment:
Not sure about this, but perhaps we can make a ticket to discuss what we want to do and then implement it. Also, there is an edge case here of if the first value is nan and if the value in filtered_trace is nan for the index before, this would copy the nan. Safer options would be to fill with either the median of the input trace or the nanmedian (or nanmean) of the filtered trace (after calculation of the median filter that is). You could hard error on if the median/mean of the full trace is NaN.

Reply:
I updated it to the linear interpolation